### PR TITLE
Spot instances support

### PR DESCRIPTION
**Pull request by stgleb**
- [ ] Fix (a **bug** or other **issue** was fixed)
- [x] Feature (**new functionality** was added)
- [ ] Breaking (**existing functionality** was affected)
- [ ] Other (please **explain** below)

Introduce spot instances to controlplane

Make controlplane able to provision AWS spot instances

Add two endpoint

POST `/kubes/{kubeID}/spot`

adds spot instance to the kube

body example

```
{
   "spotPrice": "0.0044",
   "machineType": "t2.micro",
    "machineCount": 3,
   "availabilityZone": "ap-southeast-1b"
}
```

GET `/kubes/{kubeID}/spot/{machineType}/price`

Gets spot price history for particular machine type

Response body

```
{"Prices":["0.028200","0.028200","0.028200","0.028200","0.028200","0.028200","0.028200","0.028200","0.028200"]}
```

- [x] [Documentation](https://supergiant.readme.io/v2.1.0-alpha/) was **needed**, and I updated it.
- [ ] [Documentation](https://supergiant.readme.io/v2.1.0-alpha/) was **not** needed.
- [ ] I understand [the Contribution Guidelines](https://supergiant.readme.io/docs/guidelines).

### CODE REVIEW

File: pkg/kube/util.go [hunk 1]
```diff
@@ -158,3 +164,175 @@ func syncMachines(ctx context.Context, k *model.Kube, account *model.CloudAccoun
...
+		ValidFrom:     aws.Time(time.Now().Add(time.Second * 10)),
+		// TODO(stgleb): pass this as a parameter
+		ValidUntil: aws.Time(time.Now().Add(time.Duration(24*365) * time.Hour)),
+	}
+
+	result, err := svc.RequestSpotInstances(input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			default:
```
Comment (saheienko) [inline pkg/kube/util.go]: `switch` operator has only default case. Is it required here?

Comment (stgleb) [reply pkg/kube/util.go]: No, I just copy-pasted it

File: pkg/kube/util.go [hunk 2]
```diff
@@ -158,3 +164,172 @@ func syncMachines(ctx context.Context, k *model.Kube, account *model.CloudAccoun
...
+					Key:   aws.String("KubernetesCluster"),
+					Value: aws.String(config.ClusterName),
+				},
+				{
+					Key:   aws.String(clouds.TagClusterID),
+					Value: aws.String(config.ClusterID),
+				},
+				{
+					Key: aws.String("Name"),
+					Value: aws.String(fmt.Sprintf("%s-node-%s",
```
Comment (saheienko) [inline pkg/kube/util.go]: `util.MakeNodeName` could be used here.

Comment (stgleb) [reply pkg/kube/util.go]: done

**Pull request merged**